### PR TITLE
Add io.appflowy.AppFlowy

### PR DESCRIPTION
**Pull request by LucasXu0**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LucasXu0** [comment]
bot, build io.appflowy.AppFlowy

**Comment by LucasXu0** [comment]
bot, build io.appflowy.AppFlowy

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: is this not built from source to support aarch64?

File: io.appflowy.AppFlowy.desktop [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Type=Application
+Name=AppFlowy
+Icon=io.appflowy.AppFlowy
+Exec=env GDK_GL=gles app_flowy
```
Comment (hfiguiere) [inline io.appflowy.AppFlowy.desktop:5]: this exec line is wrong. Like in the previous PR.

It should the name of the executable. Nothing else. If you need to set the env, then you should add a wrapper script

File: io.appflowy.AppFlowy.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+app-id: io.appflowy.AppFlowy
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
```
Comment (hfiguiere) [inline io.appflowy.AppFlowy.yml:3]:
```suggestion
runtime-version: '22.08'
```

File: io.appflowy.AppFlowy.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+sdk: org.freedesktop.Sdk
+command: app_flowy
+separate-locales: false
```
Comment (hfiguiere) [inline io.appflowy.AppFlowy.yml:6]: any reason to do that?

File: io.appflowy.AppFlowy.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
```
Comment (hfiguiere) [inline io.appflowy.AppFlowy.yml:14]: why "all" devices?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.appflowy.AppFlowy.desktop (hunk 2 above)
Comment (LucasXu0) [reply io.appflowy.AppFlowy.desktop]: Thanks. I have updated the code.

File: io.appflowy.AppFlowy.yml (hunk 4 above)
Comment (LucasXu0) [reply io.appflowy.AppFlowy.yml:6]: To be honest, I copied this code from another project.yml.

**Comment by LucasXu0** [comment]
bot, build io.appflowy.AppFlowy

**Comment by LucasXu0** [comment]
bot, build io.appflowy.AppFlowy

File: flathub.json (hunk 1 above)
Comment (hfiguiere) [reply flathub.json:2]: this is disappointing

File: io.appflowy.AppFlowy.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    </p>
+    <p>
+      ## Truely native experience Faster, more stable with support for offline mode. It&apos;s also better integrated with different devices. Moreover, AppFlowy enables users to access features and possibilities not available on the web.
+    </p>
+  </description>
+  
+  <launchable type="desktop-id">io.appflowy.AppFlowy.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://github.com/AppFlowy-IO/appflowy/raw/main/doc/imgs/welcome.png</image>
```
Comment (hfiguiere) [inline io.appflowy.AppFlowy.appdata.xml:37]: you link to the `main` branch which mean that when the file get removed at one point the build will break

**Review by hfiguiere — CHANGES_REQUESTED** [review]
have the desktop, appdata.xml and icon been submitted upstream? No icon or desktop file for a desktop app is a bug

File: io.appflowy.AppFlowy.appdata.xml (hunk 6 above)
Comment (LucasXu0) [reply io.appflowy.AppFlowy.appdata.xml:37]: Thanks for the reminder. We will keep this image on the main branch forever.

**Comment by LucasXu0** [comment]
bot, build io.appflowy.AppFlowy

**Comment by LucasXu0** [comment]
Hi, @hfiguiere. I got this error when I run the app using flatpak. But there is no error when running it directly.

```
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'libdart_ffi.so': libssl.so.1.1: cannot open shared object file: No such file or directory
```

It seems the app is crash when loading the dynamic library `libssl`. Could you give me some suggestions to fix it?

**Comment by hfiguiere** [comment]
I know nothing about flutter and how it build stuff. If you unpack a binary, this is the kind of stuff that can happen. The runtime 22.08 only ship with openSSL 3.

**Comment by LucasXu0** [comment]
bot, build io.appflowy.AppFlowy

**Comment by LucasXu0** [comment]
I switched the `runtime-version` to 21.08. The app works fine now. It's ready to review now.

**Comment by barthalion** [comment]
Please note that 21.08 will be supported for less than a year. You will need to upgrade to 22.08 sooner than later.

**Pull request closed without merging**